### PR TITLE
[WEB 1373] Implement out of stock items on Merlins PWA

### PR DESCRIPTION
**Pull request by tony-dinh**
Implement out of stock items on Merlins PWA

 **JIRA**: [[WEB-1373] Implement out of stock items on Merlins PWA](https://mobify.atlassian.net/browse/WEB-1373)
 **Linked PRs**: N/A

## Changes
- SFCC: Added `availability` expansion to request url to grab inventory details
- Merlin's: Parse availability information from the desktop site
- Both: Display stepper & ATC button only if item is available in stock
- Both: Add `available` to Product types

## How to test-drive this PR
- Run `npm run dev`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1):
    1. View [out of stock product](https://www.merlinspotions.com/books/necronomicon.html)
    1. Compare with any product that is in stock (like [this one](https://www.merlinspotions.com/books/beginner-s-guide-to-transfiguration-2.html))
**NOTE:** Stepper & ATC button does not render when the item is out of stock.

- Preview [DemandWare](https://preview.mobify.com/?url=https%3A%2F%2Fmobify-tech-prtnr-na03-dw.demandware.net%2Fon%2Fdemandware.store%2FSites-2017refresh-Site%2Fdefault%2FHome-Show&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1):
    1. You will notice that some of the HTTP requests used to fail; in this case it will result in the product being in an `out of stock` state. For a product where the HTTP request does succeed ([click here](https://mobify-tech-prtnr-na03-dw.demandware.net/s/2017refresh/44736828.html)). See that it works as before.

### CODE REVIEW

File: web/app/containers/product-details/partials/product-details-add-to-cart.jsx [hunk 1]
```diff
@@ -34,33 +34,37 @@ const ProductDetailsAddToCart = ({quantity, setQuantity, onSubmit, disabled, isI
...
-            <Button
-                type="submit"
-                icon="plus"
-                iconClassName="c--small u-margin-end"
-                title={isInCheckout ? 'Update Cart' : 'Add to Cart'}
-                showIconText={true}
-                className="c--primary u-width-full u-text-uppercase u-margin-bottom-lg t-product-details__add-to-cart"
-                disabled={disabled}
-            />
+            {available &&
```
Comment (jeremywiebe) [inline web/app/containers/product-details/partials/product-details-add-to-cart.jsx]: Are we going to tell the user why they aren't seeing a button when the product is out of stock?

Comment (tony-dinh) [inline web/app/containers/product-details/partials/product-details-add-to-cart.jsx]: Hmm there is supposed to be "out of stock" text displayed, so I _think_ it makes some sense for there not to be an `Add to Cart` button (the hidden button was part of the mock design).

**Comment by DanielBrookRoberge** [comment]
You should also add the availability flag to the Product type in `web/app/integration-manager/products/types.js`, since this is making that flag part of the schema that the product details page expects.

File: web/app/containers/product-details/partials/product-details-heading.jsx [hunk 2]
```diff
@@ -28,15 +28,16 @@ const ProductDetailsHeading = ({breadcrumbs, title, price, isInCheckout, cartURL
             <SkeletonBlock width="50%" height="32px" className="u-margin-bottom" />
         }
 
-        {price ?
-            <span className="t-product-details-heading__price t-product-details__price u-color-accent u-text-weight-regular u-text-family-header u-text-letter-spacing-small">{price}</span>
+        {(available != null && price != null) ?
+            price.length > 0 && <span className="t-product-details-heading__price t-product-details__price u-color-accent u-text-weight-regular u-text-family-header u-text-letter-spacing-small">{price}</span>
```
Comment (tony-dinh) [inline web/app/containers/product-details/partials/product-details-heading.jsx]: These operators ensure that we still have a skeleton so long as we are still fetching product information.

Once we have product info, if the price is unavailable we choose to not render it.

File: web/app/containers/product-details/partials/product-details-add-to-cart.jsx (hunk 1 above)
Comment (jeremywiebe) [inline web/app/containers/product-details/partials/product-details-add-to-cart.jsx]: Oops, I missed that there was an Out of Stock label.  👍   As you were.  :)

**Pull request merged**